k8s/upgrade/src/upgrade_resources/upgrade.rs

```diff
@@ -27,11 +27,28 @@ use kube::{
 };
 use std::collections::HashSet;
 
-/// The types of resources that support the upgrade operator.
+/// Arguments to be passed for upgrade.
 #[derive(clap::Subcommand, Debug)]
-pub enum UpgradeOperator {
-    /// Install, Uninstall upgrade resources.
-    UpgradeOperator,
+pub enum Upgrade {
+    /// Delete upgrade resources
+    Upgrade,
+}
+
+impl Upgrade {
+    /// Delete the upgrade resources
+    pub async fn delete(ns: &str) {
+        // Delete upgrade resources once job completes
+        match upgrade_job_completed(ns).await {
+            Ok(job_completed) => {
+                if job_completed {
+                    UpgradeResources::delete_upgrade_resources(ns).await;
+                }
```

Comment:
Maybe an else scenario for when the job is not complete is not complete is required. I feel this lacks feedback saying that the running job cannot be deleted.

Maybe also a `--force` or `-f` flag to forcibly delete.

Reply:
Force can be added in another PR.